client/src/components/ProductsandOutfits/Carousel/Carousel.jsx

```diff
@@ -7,43 +7,72 @@ const addCard = {
 };
 function Carousel({
-  products, styles, type,
+  products, styles, type, actionBtnFunc,
 }) {
   const carouselCardStyle = {
-    display: 'flex',
-    flexDirection: 'row',
-    flexWrap: 'nowrap',
-    alignItems: 'flex-end',
+    overflow: 'hidden',
   };
+
   const buttonStyle = {
-    alignSelf: 'center',
     position: 'relative',
+    top: '-290px',
   };
 
   const [current, setCurrent] = useState(0);
   const { length } = products;
-  const cardAmount = type === 'related' ? 3 : 2;
+  const cardAmount = type === 'outfit' ? length - 3 : length - 2;
+  // debugger;
+  const translateAmt = `translateX(-${current * 35}%)`;
+  const containerStyle = {
+    whiteSpace: 'nowrap',
+    transition: '0.6s',
+    transform: translateAmt,
+  };
   const next = () => {
-    setCurrent(current < length ? current + 1 : current);
+    setCurrent(current + cardAmount <= length ? current + 1 : current);
   };
   const prev = () => {
     setCurrent(current > 0 ? current - 1 : current);
   };
+  const onClickButtonHandler = (e, cardIndex, cardData) => {
+    e.stopPropagation();
+    actionBtnFunc(cardIndex, cardData);
+  };
   return (
     <div className="carousel" data-testid="Carousel" style={carouselCardStyle}>
-      <button className="button-left" type="button" onClick={prev} style={({ ...buttonStyle, left: '2%', visibility: current === 0 ? 'hidden' : 'visible' })}>{'<'}</button>
-      {type === 'outfit' && <ProductCards card={addCard.product} style={addCard.style} clickFunc={() => console.log('hi')} />}
-      {products.map((p, i) => {
-        const filter = styles[i].results.filter((style) => style['default?'] === true);
-        const style = filter.length ? filter[0] : styles[i].results[0];
-        const isVisible = (i >= current && i < current + cardAmount);
-        return (
-          <div className="Card" style={{ contentVisibility: isVisible ? 'visible' : 'hidden' }} key={style.style_id}>
-            <ProductCards key={style.style_id} card={p} style={style} clickFunc={(card) => console.log('cardClicked\n', 'ProductData:', card, '\nStyleData:', style)} />
-          </div>
-        );
-      })}
-      <button className="button-right" type="button" onClick={next} style={({ ...buttonStyle, right: '2%', visibility: current + cardAmount >= length ? 'hidden' : 'visible' })}>{'>'}</button>
+      <div className="container" data-testid="Container" style={containerStyle}>
+
+        {type === 'outfit' && <ProductCards card={addCard.product} style={addCard.style} actionButton={() => console.log('outfit action button clicked')} clickFunc={() => console.log('hi')} />}
+        {products.map((p, i) => {
+          const filter = styles[i].results.filter((style) => style['default?'] === true);
+          const style = filter.length ? filter[0] : styles[i].results[0];
```

Comment:
filter and style here could have better variable names. Only pointing this out cause otherwise I wouldn't have anything else to point out 😅 Looks really good!